### PR TITLE
In place batch_norm

### LINKED ISSUE #9416: In-place Operations
One of the low hanging fruits in memory optimization is in-place operators. If you look at the [proto message](http://ethereon.github.io/netscope/#/gist/3df1f508f14651cf127e9fbf009453ca) of ResNet, it has tons of in-place operations: conv's output has been in place modified by batch_norm, scale, and Relu.

With a few lines modification of code, my experiments showed that it can save up to 40% of the memory.

|                            | VGG  | ResNet50 |
|----------------------------|------|----------|
| Original                   | 3.3G | 1.5G     |
| In-place BatchNorm         | 2.9G | 1.2G     |
| In-place BN and activation | 2.5G | 0.9G     |
| memopt | 2.4G | 0.94G |
| memopt + in-place | 2.0G | 0.6G|

### PR DESCRIPTION
**Pull request by tonyyang-svail**
fix #9416
1. add in-place attr to bn
1. make batch norm in-place in img_conv_group by default

### CODE REVIEW

**Comment by jacquesqiao** [comment]
have you verified the correctness of inplace for batch_nrom? I looked at the code and think it has no problem.
